src/unwinder/arch/mips.rs

```diff
@@ -0,0 +1,397 @@
+            #[cfg(target_feature = "single-float")]
+            Register(32..=63) => &mut self.fp[(reg.0 - 32) as usize],
+            _ => unimplemented!(),
+        }
+    }
+}
+
+macro_rules! code {
+    (save_gp) => {
+        "
+        sw $zero,   0x00($sp)
+        sw $ra,     0x7C($sp)
+        sw $gp,     0x70($sp)
+        sw $v0,     0x08($sp)
+        sw $v1,     0x0C($sp)
+        sw $a0,     0x10($sp)
+        sw $a1,     0x14($sp)
+        sw $a2,     0x18($sp)
+        sw $a3,     0x1C($sp)
+        sw $t0,     0x20($sp)
+        sw $t1,     0x24($sp)
+        sw $t2,     0x28($sp)
+        sw $t3,     0x2C($sp)
+        sw $t4,     0x30($sp)
+        sw $t5,     0x34($sp)
+        sw $t6,     0x38($sp)
+        sw $t7,     0x3C($sp)
+        sw $s0,     0x40($sp)
+        sw $s1,     0x44($sp)
+        sw $s2,     0x48($sp)
+        sw $s3,     0x4C($sp)
+        sw $s4,     0x50($sp)
+        sw $s5,     0x54($sp)
+        sw $s6,     0x58($sp)
+        sw $s7,     0x5C($sp)
+        sw $t8,     0x60($sp)
+        sw $t9,     0x64($sp)
+        "
+    };
+    (save_fp) => {
+        "
+        swc1 $f0,   0x80($sp)
+        swc1 $f1,   0x84($sp)
+        swc1 $f2,   0x88($sp)
+        swc1 $f3,   0x8C($sp)
+        swc1 $f4,   0x90($sp)
+        swc1 $f5,   0x94($sp)
+        swc1 $f6,   0x98($sp)
+        swc1 $f7,   0x9C($sp)
+        swc1 $f8,   0xA0($sp)
+        swc1 $f9,   0xA4($sp)
+        swc1 $f10,  0xA8($sp)
+        swc1 $f11,  0xAC($sp)
+        swc1 $f12,  0xB0($sp)
+        swc1 $f13,  0xB4($sp)
+        swc1 $f14,  0xB8($sp)
+        swc1 $f15,  0xBC($sp)
+        swc1 $f16,  0xC0($sp)
+        swc1 $f17,  0xC4($sp)
+        swc1 $f18,  0xC8($sp)
+        swc1 $f19,  0xCC($sp)
+        swc1 $f20,  0xD0($sp)
+        swc1 $f21,  0xD4($sp)
+        swc1 $f22,  0xD8($sp)
+        swc1 $f23,  0xDC($sp)
+        swc1 $f24,  0xE0($sp)
+        swc1 $f25,  0xE4($sp)
+        swc1 $f26,  0xE8($sp)
+        swc1 $f27,  0xEC($sp)
+        swc1 $f28,  0xF0($sp)
+        swc1 $f29,  0xF4($sp)
+        swc1 $f30,  0xF8($sp)
+        swc1 $f31,  0xFC($sp)
+        "
+    };
+    (restore_gp) => {
+        "
+        lw $ra,     0x7C($a0)
+        lw $gp,     0x70($a0)
+        lw $v0,     0x08($a0)
+        lw $v1,     0x0C($a0)
+        lw $a1,     0x14($a0)
+        lw $a2,     0x18($a0)
+        lw $a3,     0x1C($a0)
+        lw $t0,     0x20($a0)
+        lw $t1,     0x24($a0)
+        lw $t2,     0x28($a0)
+        lw $t3,     0x2C($a0)
+        lw $t4,     0x30($a0)
+        lw $t5,     0x34($a0)
+        lw $t6,     0x38($a0)
+        lw $t7,     0x3C($a0)
+        lw $s0,     0x40($a0)
+        lw $s1,     0x44($a0)
+        lw $s2,     0x48($a0)
+        lw $s3,     0x4C($a0)
+        lw $s4,     0x50($a0)
+        lw $s5,     0x54($a0)
+        lw $s6,     0x58($a0)
+        lw $s7,     0x5C($a0)
+        lw $t8,     0x60($a0)
+        lw $t9,     0x64($a0)
```

Comment:
Where is the restoring code for at, k0, k1, fp, sp? Also please arrange everything in offset order.

Reply:
All non-volatile registers need to be saved, and all registers need to be restored.